crates/criticalup-core/src/download_server_client.rs

```diff
@@ -54,18 +54,11 @@ impl DownloadServerClient {
 
     #[tracing::instrument(level = "debug", skip_all)]
     pub async fn get_keys(&self) -> Result<Keychain, Error> {
-        let mut keychain = Keychain::new(&self.trust_root).map_err(Error::KeychainInitFailed)?;
-
         let resp: KeysManifest = self
             .json(self.send(self.client.get(self.url("/v1/keys"))).await?)
             .await?;
-        for key in &resp.keys {
-            // Invalid keys are silently ignored, as they might be signed by a different root key
```

Comment:
This note might be good to add to `load_all`.

Reply:
Preserved.